File: src/models/subject.go
package models

// Subject represents a single subject that can be voted on
type Subject struct {
	Model
	PersonName         string           `json:"personName" binding:"required" gorm:"not null"`
	CostumeDescription string           `json:"costumeDescription" binding:"required" gorm:"not null"`
	Votes              []Vote           `json:"votes"`
	History            []SubjectHistory `json:"history"`
	VoteCount          int              `gorm:"-" json:"voteCount"`
}

File: graphql-server/src/db/models/subject.js
'use strict';
module.exports = (sequelize, DataTypes) => {
  var subject = sequelize.define(
    'subject',
    {
      personName: DataTypes.STRING,
      costumeDescription: DataTypes.STRING,
    },
    {},
  );
  subject.associate = function(models) {
    // associations can be defined here
    subject.hasMany(models.subject_history, {
      as: 'history',
      foreignKey: 'subjectId',
    });
    subject.hasMany(models.vote, { as: 'votes', foreignKey: 'subjectId' });
  };
  return subject;
};
